From 7ee50b2dede7a7c4551ac2c67aa7409f55dc1fc2 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:03:42 -0400
Subject: [usb] wusbcore: fix control-pipe directions

Message-id: <1410887463-20674-151-git-send-email-dzickus@redhat.com>
Patchwork-id: 95377
O-Subject: [RHEL7 PATCH 150/591] USB: wusbcore: fix control-pipe directions
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit df3ce23a56654bbf08e0c36ab1690cd8316a53c4
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:52 2014 +0200

    USB: wusbcore: fix control-pipe directions

    Fix incorrect pipe directions in control requests (which has been
    silently fixed up by USB core).

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/wusbcore/wa-rpipe.c b/drivers/usb/wusbcore/wa-rpipe.c
index 91422c8..acd64cc 100644
--- a/drivers/usb/wusbcore/wa-rpipe.c
+++ b/drivers/usb/wusbcore/wa-rpipe.c
@@ -525,7 +525,7 @@ void rpipe_ep_disable(struct wahc *wa, struct usb_host_endpoint *ep)
   u16 index = le16_to_cpu(rpipe->descr.wRPipeIndex);
 
   usb_control_msg(
-   wa->usb_dev, usb_rcvctrlpipe(wa->usb_dev, 0),
+   wa->usb_dev, usb_sndctrlpipe(wa->usb_dev, 0),
    USB_REQ_RPIPE_ABORT,
    USB_DIR_OUT | USB_TYPE_CLASS | USB_RECIP_RPIPE,
    0, index, NULL, 0, USB_CTRL_SET_TIMEOUT);
@@ -546,7 +546,7 @@ void rpipe_clear_feature_stalled(struct wahc *wa, struct usb_host_endpoint *ep)
   u16 index = le16_to_cpu(rpipe->descr.wRPipeIndex);
 
   usb_control_msg(
-   wa->usb_dev, usb_rcvctrlpipe(wa->usb_dev, 0),
+   wa->usb_dev, usb_sndctrlpipe(wa->usb_dev, 0),
    USB_REQ_CLEAR_FEATURE,
    USB_DIR_OUT | USB_TYPE_CLASS | USB_RECIP_RPIPE,
    RPIPE_STALL, index, NULL, 0, USB_CTRL_SET_TIMEOUT);
-- 
1.7.1